.github/workflows/semver-checks.yml

```diff
@@ -22,15 +22,24 @@ jobs:
         run: cargo binstall cargo-semver-checks --no-confirm
       - name: "Run semver checker script"
         run: ./contrib/check-semver.sh
-      - name: "Add PR label to breaking changes"
-        uses: actions-ecosystem/action-add-labels@v1
+      - name: "Add PR label/comment announcing breaking changes"
+        uses: actions/github-script@v7
         if: ${{ hashFiles('semver-break') != '' }}
+        env:
+          PR_NUMBER: ${{ github.event.pull_request.number }}
         with:
-          labels: "API break"
-      - name: Comment PR
-        uses: thollander/actions-comment-pull-request@v2
-        if: ${{ hashFiles('semver-break') != '' }}
-        with:
-          message: |
-            :rotating_light: API BREAKING CHANGE DETECTED
+          github-token: ${{ secrets.APOELSTRA_CREATE_PR_TOKEN }}
```

Comment:
Is this really available in `pull_request` ccontext? Isn't it limited to `pull_request_target`?

Reply:
Damn! You're probably right.

From https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#controlling-changes-from-forks-to-workflows-in-public-repositories

> workflows from forks do not have access to sensitive data such as secrets

Reply:
You definitely need to do the two-job trick.

Reply:
Yes, I'm definitely convinced now, see https://github.com/actions/github-script/issues/370